src/ja/server/scheduler/scheduler.py

```diff
@@ -76,4 +76,7 @@ def reschedule(self, database: ServerDatabase) -> None:
         for machine in available_machines:
             database.update_work_machine(machine)
 
-        self._dispatcher.set_distribution(new_schedule)
+        lost_wms: List[WorkMachine] = self._dispatcher.set_distribution(new_schedule)
```

Comment:
Since this code isn't easily tested, I think you should add at least a unit test. Should be simple - you need:

1. A work machine with 1 job running
2. Mock the dispatcher.set_distribution function to return exactly this work machine
3. Verify that the machine is offline and that the job is crashed